blockchain/cryptoProvider/cryptoComponentsHolder.go

```diff
@@ -2,41 +2,33 @@ package cryptoProvider
 
 import (
 	crypto "github.com/ElrondNetwork/elrond-go-crypto"
-	"github.com/ElrondNetwork/elrond-go-crypto/signing"
-	"github.com/ElrondNetwork/elrond-go-crypto/signing/ed25519"
 	"github.com/ElrondNetwork/elrond-sdk-erdgo/core"
 	"github.com/ElrondNetwork/elrond-sdk-erdgo/data"
 )
 
-var (
-	suite  = ed25519.NewEd25519()
-	keyGen = signing.NewKeyGenerator(suite)
-)
-
 type cryptoComponentsHolder struct {
 	publicKey      crypto.PublicKey
 	privateKey     crypto.PrivateKey
 	addressHandler core.AddressHandler
+	bech32Address  string
 }
 
 // NewCryptoComponentsHolder returns a new cryptoComponentsHolder instance
-func NewCryptoComponentsHolder(skBytes []byte) (*cryptoComponentsHolder, error) {
+func NewCryptoComponentsHolder(keyGen crypto.KeyGenerator, skBytes []byte) (*cryptoComponentsHolder, error) {
 	privateKey, err := keyGen.PrivateKeyFromByteArray(skBytes)
 	if err != nil {
 		return nil, err
 	}
 	publicKey := privateKey.GeneratePublic()
-
-	publicKeyBytes, err := publicKey.ToByteArray()
-	if err != nil {
-		return nil, err
-	}
+	publicKeyBytes, _ := publicKey.ToByteArray()
```

Comment:
I know this can not return an error, as the current implementations look. However, masking this can lead to some unexpected behavior. Suggestion to revert the changes here.